packages/ses/src/compartment-shim.js

```diff
@@ -120,6 +127,32 @@ export const CompartmentPrototype = {
     return weaksetHas(knownScopeProxies, value);
   },
 
+  /**
+   * @param {Object} [options]
+   * @param {boolean} [options.sloppyGlobalsMode]
+   * @param {Object} [options.__moduleShimLexicals__]
+   */
+  /* eslint-disable-next-line no-underscore-dangle */
+  __makeScopeProxy__(options = {}) {
+    const compartmentFields = weakmapGet(privateFields, this);
+    const { globalObject, knownScopeProxies } = compartmentFields;
+
+    const { sloppyGlobalsMode } = options;
+    const localObject = makeCompartmentLocalObject(compartmentFields, options);
+    // here we drop "admitOneUnsafeEvalNext" and "resetOneUnsafeEvalNext"
+    // as we never enable eval for this handler
+    const { scopeHandler } = createScopeHandler(globalObject, localObject, {
+      sloppyGlobalsMode,
+    });
+    const { proxy: scopeProxy, revoke: revokeScopeProxy } = proxyRevocable(
+      immutableObject,
+      scopeHandler,
+    );
+    weaksetAdd(knownScopeProxies, scopeProxy);
```

Comment:
This line supports your intuition to make this a method of compartment, provided that `__knownScopeProxies__` can possibly interact with `__makeScopeProxy__`. I am not sure that possibility exists if the intention is to use the scope proxy instead of the compartment evaluator.

Reply:
> I am not sure that possibility exists if the intention is to use the scope proxy instead of the compartment evaluator.

should be no detectable difference in behavior vs the compartment evaluator. eg can still leak the scopeProxy